CONTRIBUTING.md

```diff
@@ -0,0 +1,121 @@
+# Contributing to ocaml-lsp
+
+OCaml-lsp is a community oriented open-source project and we encourage and value
+any kind of contribution. Thanks for taking the time to contribute 🐫 !
+
+## Code of Conduct
+
+OCaml-lsp  adheres to the OCaml Code of Conduct as stated in the [Code of Conduct
+document](CODE_OF_CONDUCT.md). By participating, you are expected to uphold this
+code. Please report unacceptable behavior either to local contacts (listed in
+[here](CODE_OF_CONDUCT.md)) or to someone listed in the upstream [OCaml Code of
+Conduct](CODE_OF_CONDUCT.md).
+
+## Documentation
+
+Much of the information relating to the repository, such as installation
+guidelines, how to set up a development environment and how to run unit tests,
+can be found in the project [README](README.md). And custom requests are
+documented in the
+[ocaml-lsp-server/docs/ocamllsp](ocaml-lsp-server/docs/ocamllsp) directory.
+
+Contributions to the documentation are welcome!
+
+
+## Question, bug reports and feature requests
+
+We rely on [Github's issue tracker](https://github.com/ocaml/ocaml-lsp/issues) for
+support questions, feature requests and bug reports.
+
+When reporting an issue, please include a precise reproduction in the bug report
+when that's possible, as it is a very useful tool to investigate. You should
+also check that you are using the latest version of OCaml-lsp and that a similar
+issue has not already been submitted.
+
+## Code contributions
+
+### Styleguides
+
+- **OCaml**: a large part of the code base is written in OCaml and the project
+  is configured to work with
+  [ocamlformat](https://ocaml.org/p/ocamlformat/latest) (version defined in the
+  [`.ocamlformat` file](.ocamlformat)).
+
+- **TypeScript**: TypeScript is used to describe certain end-to-end tests
+  (abbreviated as `e2e`) and the project uses the
+  [prettier](https://prettier.io/) formatter.
```

Comment:
Would be good to mention that these tests are deprecated and we do not allow extending them anymore. Gradually we'll rewrite them all to OCaml.